clarity_ext/domain/analyte.py

```diff
@@ -47,9 +48,14 @@ def create_from_rest_resource(resource, is_input, udf_map, container_repo):
         # TODO: sample should be put in a lazy property, and all samples in a step should be
         # loaded in one batch
         samples = [Sample.create_from_rest_resource(sample) for sample in resource.samples]
+        is_control = False
+        # TODO: This code principally belongs to the genologics layer, but 'control-type' does not exist there
+        if resource.root.find("control-type") is not None:
```

Comment:
Ahh. OK. I don't have access to that URL at the moment, but if `root` is an XML property that always exists, and `find()` returns `None` when an element is not found (instead of with strings when it would be `-1` e.g.), then I guess I'm happy. :smile: